explore/src/main/scala/explore/observationtree/ObsSummaryTable.scala

```diff
@@ -134,44 +159,134 @@ object ObsSummaryTable extends TableHooks:
 
         List(
           // TODO: GroupsColumnId
-          column(ObservationIdColumnId, _.id).sortable,
+          column(ObservationIdColumnId, _.obs.id),
 
           // TODO: ValidationCheckColumnId
-          column(StatusColumnId, _.status),
+          column(StatusColumnId, _.obs.status),
           // TODO: CompletionColumnId
           // TODO: TargetTypeColumnId
+          ColDef(
+            ColumnId("expander"),
+            cell = cell =>
+              if (cell.row.getCanExpand())
+                <.span(
+                  ^.cursor.pointer,
+                  ExploreStyles.ExpanderChevron,
+                  ExploreStyles.ExpanderChevronOpen.when(cell.row.getIsExpanded()),
+                  ^.onClick ==> (_.stopPropagationCB *> Callback(
+                    cell.row.getToggleExpandedHandler()()
+                  ))
+                )(Icons.ChevronRightLight.withFixedWidth(true))
+              else "",
+            size = 18.toPx,
+            enableResizing = false
+          ),
           column(TargetTypeColumnId, _ => ())
             .setCell(_ => Icons.Star.withFixedWidth())
             .setSize(35.toPx),
-          column(TargetColumnId, _.title),
-          column(ConstraintsColumnId, obs => (obs.id, obs.constraintsSummary)).setCell(cell =>
-            val (id, constraintsSummary) = cell.value
-            <.a(^.href := constraintUrl(id),
-                ^.onClick ==> (_.preventDefaultCB *> goToConstraint(id)),
-                constraintsSummary
+          column(TargetColumnId, identity, identity).setCell { cell =>
+            def targetUrl(obsId: Observation.Id, tWId: TargetWithId) = <.a(
+              ^.href := ctx.pageUrl(AppTab.Targets, props.programId, Focused.target(tWId.id)),
+              ^.onClick ==> (e =>
+                e.preventDefaultCB *> e.stopPropagationCB *> ctx.pushPage(
+                  AppTab.Observations,
+                  props.programId,
+                  Focused.singleObs(obsId, tWId.id.some)
+                )
+              ),
+              tWId.target.name.value
+            )
+
+            cell.value match {
+              case ExpandedTargetRow(obsId, target) => targetUrl(obsId, target)
+              case ObsRow(obs, _, _)                => obs.title
+            }
+          },
+          column(ConstraintsColumnId, r => (r.obs.id, r.obs.constraintsSummary)).setCell(cell =>
+            cell.value.map((id, constraintsSummary) =>
+              <.a(^.href := constraintUrl(id),
+                  ^.onClick ==> (_.preventDefaultCB *> goToConstraint(id)),
+                  constraintsSummary
+              )
             )
           ),
           // TODO: FindingChartColumnId
-          column(ConfigurationColumnId, _.conf),
-          column(DurationColumnId, _.executionTime.toHoursMinutes)
+          column(ConfigurationColumnId, _.obs.conf),
+          column(DurationColumnId, _.obs.executionTime.toHoursMinutes),
```

Comment:
Is the executionTime already in the model?

Reply:
It's returned from the backend query, yes